scripts/globals/beastmentreasure.lua

```diff
@@ -0,0 +1,394 @@
+------------------------------------
+-- Beastmen Treasure side quests
+-- https://ffxiclopedia.fandom.com/wiki/Beastmen_Treasure
+------------------------------------
+require('scripts/globals/keyitems')
+require('scripts/globals/quests')
+require('scripts/globals/status')
+require('scripts/globals/weather')
+------------------------------------
+dsp = dsp or {}
+dsp.beastmentreasure = dsp.beastmentreasure or {}
+
+local PICKAXE = 605
+
+local zoneData = {
+    -- Yuhtunga Jungle (Elshimo Lowlands)
+    [123] =
+    {
+        statusvar = 'BMT_Lowlands_Status',
+        qmvar = 'BMT_Lowlands_DigQM',
+        mapid = dsp.ki.MAP_OF_THE_ELSHIMO_REGIONS,
+        peddlestox =
+        {
+            id = 17281639,
+            day = dsp.day.WINDSDAY
+        },
+        fetchitems =
+        {
+            1480, -- Mermaid Head
+            1481, -- Mermaid Body
+            1482, -- Mermaid Hands
+            1483  -- Mermaid Tail
+        },
+        digsites =
+        {
+            [3]  = {id = 17281642, x = 83.726,   y = 448.329,  chest = 0},
```

Comment:
> No ID's in anything but IDs.lua

Ok. I'll look into refactoring NPC ids to the zone-specific `IDs.lua` files.

> There is no reason to keep the positions in this table, just use entity:getPos() and use the values in the table returned.

Trying to minimize the amount of db calls that need to be done at runtime. Since we're already loading a global, we have the option to not run any.

> Why not just have the key value be the chest number? We don't care about which qm# it is in our database.

Originally I set it up this way so that dig sites could be added as they were captured without breaking existing player vars (in case a dig site were added out of order.) We only had one verified dig site at this point, and we didn't know about the chest animation args for event 105. After figuring out the rest of the dig sites and their chest args I considered changing it to what you're proposing, but I opted to leave it alone for a few reasons: 

- We need to store which chest is assigned to the player, and if the player is ever assigned chest 0 then `setVar()` will simply delete the variable. We could shift them all up by one when stored and shift them back down when read, but that gets a little confusing.
- There's a meaningful relationship between qm# and "chest arg" that we have no way to look up anywhere else. If someone is testing or troubleshooting and they see that a player has been assigned qm7, you know exactly what that means and how to get there. If instead they've been assigned *chest 7*, then you're looking through the code to figure out that chest 7 actually corresponds to chest 6, and what even is "chest 6"? Now, once the IDs are refactored to `IDs.lua` I guess you could follow that, but it still seems more intuitive to think of these in terms of QM == dig site and "chest" is just an arg associated with that dig site, rather than event arg == NPC id.
- Lua has no `table.indexOf()` so I think we're stuck iterating either way.

Reply:
> Trying to minimize the amount of db calls that need to be done at runtime. Since we're already loading a global, we have the option to not run any.

position is in memory after server start, its not a db call.

Reply:
> 
> 
> > Trying to minimize the amount of db calls that need to be done at runtime. Since we're already loading a global, we have the option to not run any.
> 
> position is in memory after server start, its not a db call.

Whoops. In that case I think I agree, coordinates can be dropped here in favor of `getPos()`.

Reply:
> Lua has no table.indexOf() so I think we're stuck iterating either way.

not saying it is or isn't worth doing here (I haven't even read what all you did yet), but its possible to build an inverse saved for later if that is needed frequently.

Reply:
> No ID's in anything but IDs.lua
> There is no reason to keep the positions in this table, just use entity:getPos() and use the values in the table returned.
> Why not just have the key value be the chest number? We don't care about which qm# it is in our database.

Implemented all of these suggestions in https://github.com/DarkstarProject/darkstar/commit/d5d980bf581fddc0fefaa916eb683b99ca748faf.

The only potential concern I have is the order that digsites are listed in in `npc.BEASTMEN_TREASURE` in `IDs.lua` is important. If it's possible that an update script might re-order them I'll go ahead and define explicit keys, but for now chest number is just implied by the index.